.github/workflows/test-and-publish.yml

```
@@ -0,0 +1,59 @@
+name: Test
+
+on:
+  push:
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Run unit tests
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./run-tests.sh
+  api-tests:
+    name: API Tests
+    runs-on: ubuntu-22.04
```

I believe we've decided to just use latest for all these, since we didn't have control over this before either, correct me if I'm wrong.

```suggestion
    runs-on: ubuntu-latest
```